src/theia/matching/flann_feature_matcher.cc

```diff
@@ -86,17 +87,50 @@ bool FlannFeatureMatcher::MatchImagePair(
 }
 
 void FlannFeatureMatcher::AddImage(
+    const std::string& image_name,
+    const std::vector<Keypoint>& keypoints,
+    const std::vector<Eigen::VectorXf>& descriptors,
+    const CameraIntrinsicsPrior& intrinsics) {
+  AddImage(image_name, keypoints, descriptors);
+  std::lock_guard<std::mutex> lock(indexed_images_lock_);
```

Comment:
Can you add to `indexed_images_lock_` description that it protect also `intrinsics_` and `image_names_`